pom.xml

```diff
@@ -1605,6 +1605,260 @@
+                          <minimum>0.96</minimum>
+                        </limit>
+                        <limit>
+                          <counter>BRANCH</counter>
+                          <value>COVEREDRATIO</value>
+                          <minimum>0.73</minimum>
+                        </limit>
+                      </limits>
+                    </rule>
+                    <rule>
+                      <element>CLASS</element>
+                      <includes>
+                        <include>com.puppycrawl.tools.checkstyle.grammars.GeneratedJavaLexer</include>
+                      </includes>
+                      <limits>
+                        <limit>
+                          <counter>LINE</counter>
+                          <value>COVEREDRATIO</value>
+                          <minimum>0.96</minimum>
+                        </limit>
+                        <limit>
+                          <counter>BRANCH</counter>
+                          <value>COVEREDRATIO</value>
+                          <minimum>0.79</minimum>
+                        </limit>
+                      </limits>
+                    </rule>
+                    <rule>
+                      <element>CLASS</element>
+                      <includes>
+                        <include>com.puppycrawl.tools.checkstyle.grammars.javadoc.JavadocParser</include>
+                      </includes>
+                      <limits>
+                        <limit>
+                          <counter>LINE</counter>
+                          <value>COVEREDRATIO</value>
+                          <minimum>0.54</minimum>
+                        </limit>
+                        <limit>
+                          <counter>BRANCH</counter>
+                          <value>COVEREDRATIO</value>
+                          <minimum>0.44</minimum>
+                        </limit>
+                      </limits>
+                    </rule>
+                    <rule>
+                      <element>CLASS</element>
+                      <includes>
+                        <include>com.puppycrawl.tools.checkstyle.grammars.javadoc.JavadocLexer</include>
+                      </includes>
+                      <limits>
+                        <limit>
+                          <counter>LINE</counter>
+                          <value>COVEREDRATIO</value>
+                          <minimum>0.84</minimum>
+                        </limit>
+                        <limit>
+                          <counter>BRANCH</counter>
+                          <value>COVEREDRATIO</value>
+                          <minimum>0.64</minimum>
+                        </limit>
+                      </limits>
+                    </rule>
+                    <rule>
+                      <element>CLASS</element>
+                      <includes>
+                        <include>com.puppycrawl.tools.checkstyle.checks.javadoc.ClassResolver</include>
```

Comment:
Is this class truly not 100%, along with the others below?
Everything but the ANTLR should be 100%. Why isn't cobertura reporting these missing coverages?

Reply:
In the google groups there is a topic about this:
https://groups.google.com/d/msg/checkstyle-devel/KnPfFZZiumA/mp2pNzFNBwAJ

In short, the Jacoco heuristics does not detect methods that never return, while the Cobertura heuristics does it.

All of these classes:
`ClassResolver`,` JavadocUtils`, `PackageNamesLoader`,` Main`, `JavadocDetailNodeParser.JavadocParserErrorStrategy` can be fixed to get 100% coverage by some hacks and workarounds. But this is beyond the scope of this PR.

Reply:
Thanks, I thought I saw the reason somewhere but I couldn't remember.

> the Jacoco heuristics does not detect methods that never return

Has Jacoco stated if they will ever change this?